src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -105,7 +105,10 @@ void ExecuteWithAbi (string supportedAbis, string apkInputPath, string apkOutput
 						Assembly.GetExecutingAssembly ().GetManifestResourceStream ("NOTICE.txt"));
 
 				// Add classes.dx
-				apk.Archive.AddFiles (DalvikClasses, useFileDirectories: false);
+				foreach (var dex in DalvikClasses) {
+					string apkName = dex.GetMetadata ("ApkName");
```

Comment:
What is this `%(ApkName)` metadata?  `git grep 'ApkName'` doesn't find anything, and I don't see anything within this PR which would set that metadata.

Reply:
https://github.com/xamarin/monodroid/pull/929/files#diff-fb29880b2363fa3fda1824e34aaecd51R230

Reply:
This is used to change the name of the file when its added to the apk. Android requires a classes.dex to be present, so we can't just use the actual name of the file.

Reply:
Could we add some "documentation" for the `%(ApkName)` metadata?  Not necessarily within `Documentation`, but something to "call out" within the commit message.